cdk/src/open-data-platform/data-plane/data-import/write-demographic-data.handler.ts

```diff
@@ -1,149 +1,211 @@
+  const batchExecuteParams: BatchExecuteStatementRequest = {
+    database: process.env.DATABASE_NAME ?? 'postgres',
+    parameterSets: rows,
+    resourceArn: process.env.RESOURCE_ARN ?? '',
+    schema: 'public',
+    secretArn: process.env.CREDENTIALS_SECRET ?? '',
+    sql: `INSERT INTO demographics (census_geo_id, total_population,
+                                    under_five_population, poverty_total,
+                                    black_percentage,
+                                    white_percentage, geom)
+          VALUES (:census_geo_id,
+                  :total_population,
+                  :under_five_population,
+                  :poverty_total,
+                  :black_percentage,
+                  :white_percentage,
+                  ST_AsText(ST_GeomFromGeoJSON(:geom))) ON CONFLICT (census_geo_id) DO NOTHING`,
+  };
+  return rdsService.batchExecuteStatement(batchExecuteParams).promise();
+}
 
-  return db.query(sql`INSERT INTO demographics (census_geo_id, total_population,
-                                                black_percentage,
-                                                white_percentage, geom)
-                      VALUES ${sql.join(
-                        rows.map((row: DemographicsTableRow) => {
-                          return sql`(
-                                         ${row.census_geo_id}, ${row.total_population},
-                                         ${row.black_percentage},
-                                         ${row.white_percentage},
-                                         ${geometry})`;
-                        }),
-                        ',',
-                      )};`);
+/**
+ * Sometimes these fields are negative because they are based on a regression.
+ */
+function getValueOrDefault(field: string): number {
+  return Math.max(parseFloat(field == 'NaN' || field == null ? '0' : field), 0);
 }
 
 /**
- * Inserts all rows into the demographics table.
+ * Maps a data row to a table row ready to write to the db.
+ * @param row: row with all data needed to build a [WaterSystemsTableRow].
  */
-async function deleteRows(db: Queryable): Promise<any[]> {
-  return db.query(sql`DELETE
-                      FROM demographics
-                      WHERE census_geo_id IS NOT NULL`);
+function getTableRowFromRow(row: any): SqlParametersList {
+  const value = row.value;
+  const properties = value.properties;
+
+  // TODO(kailamjeter): modify field names when file is fixed.
+  return (
+    new DemographicsTableRowBuilder()
+      .censusGeoId(properties.GEOID)
+      .underFivePopulation(getValueOrDefault(properties.age_under5))
+      .povertyTotal(getValueOrDefault(properties.PovertyTot))
+      .totalPopulation(getValueOrDefault(properties.RaceTotal))
+      .blackPopulation(getValueOrDefault(properties.Estimate_1))
+      .whitePopulation(getValueOrDefault(properties.Estimate!!)) // This converts to 'Estimate' when minified so this is always null as of now.
```

Comment:
Can this reference white/black_population or are you going to do a separate PR that uses the new files

Update: saw the later TODO that will use the new files

Reply:
I think this is an older commit? looking at code changes in my view I see the new fields - can you double check you are looking at the total code changes and just the first commit?